types/payment_voucher.go

```diff
@@ -74,3 +75,12 @@ func (voucher *PaymentVoucher) Encode() (string, error) {
 
 	return multibase.Encode(multibase.Base58BTC, cborVoucher)
 }
+
+// SortVouchersByValidAt takes a list of payment vouchers and sorts them
+// in increasing value of ValidAt (block height)
+func SortVouchersByValidAt(sorted []*PaymentVoucher) []*PaymentVoucher {
```

Comment:
This is part of the ask for the issue:  that the vouchers be sorted in increasing ValidAt block heights, and then indexed, and the index should be output as well.